### PR TITLE
[Non Upstream Patches] Update Makefile to include non-upstream patches

### PR DESCRIPTION
**Pull request by vivekrnv**
Signed-off-by: Vivek Reddy <vkarri@nvidia.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -99,6 +100,26 @@ $(addprefix $(DEST)/, $(MAIN_TARGET)): $(DEST)/% :
...
+	mkdir $(NON_UP_LOC)
+
+	if [ ! -z ${EXTERNAL_KERNEL_PATCHES} ];  then
+		wget ${EXTERNAL_KERNEL_PATCHES} -O patches.tar.gz
+		tar -zxf patches.tar.gz -C $(NON_UP_LOC)
+	fi
+
+	# Precedence is given for external URL
+	if [ -z ${EXTERNAL_KERNEL_PATCHES} ] && [ x${INCLUDE_MLNX_PATCHES} == xy ]; then
+		if [ -f "../../../$(MLNX_PATCH_LOC)" ]; then
```
Comment (stepanblyschak) [inline Makefile]: Can MLNX_PATCH_LOC be an absolute path ?

File: Makefile [hunk 2]
```diff
@@ -99,6 +100,26 @@ $(addprefix $(DEST)/, $(MAIN_TARGET)): $(DEST)/% :
 	stg repair
 	stg import -s ../patch/series
 
+	mkdir $(NON_UP_LOC)
```
Comment (stepanblyschak) [inline Makefile:103]: rm -rf $(NON_UP_LOC) before this line to start from scratch

Comment (vivekrnv) [reply Makefile:103]: `rm -rf $(BUILD_DIR)` is removed during the start of the build, NON_UP_LOC is present under BUILD_DIR, so this isn't required

File: Makefile (hunk 1 above)
Comment (vivekrnv) [reply Makefile]: Updated, will be set in `platform/mellanox/rules.mk`

**Review by stepanblyschak — APPROVED** [review]

**Pull request closed without merging**